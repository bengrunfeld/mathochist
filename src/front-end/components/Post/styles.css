.post h1, .post h2, .post h3 {
  font-family: comfortaa, helvetica;
}

.post h4, .post p {
  font-family: nunito, helvetica;
}

.post h1 {
  margin-bottom: 25px;
}

.post h2, .post h3 {
  margin-top: 50px;
  margin-bottom: 15px;
}

.post h2 {
  font-size: 1.5rem;
}

.post h3 {
  font-size: 1.25rem;
}

.post pre {
  padding:25px;
  background: #e3e3e3;
}