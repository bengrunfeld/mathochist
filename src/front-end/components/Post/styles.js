import styled from 'styled-components'


const PostText = styled.div`
  font-family: nunito, helvetica;
`

export {
  PostText
}