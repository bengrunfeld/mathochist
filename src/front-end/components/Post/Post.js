import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import ReactMarkdown from 'react-markdown'

import './styles.css'


const sendRequest = async(postId, updateState) => 
  await fetch(`/api/posts/${postId}`) // eslint-disable-line no-undef  
    .then(resp => resp.json())
    .then(data => updateState(data))

const Post = ({postId}) => {
  const [state, setState] = useState('')

  // The function in useState only fires on the initial render.
  useEffect(() => {
    sendRequest(postId, setState)
  }, [postId])

  return (
    <div className='post col-lg-10'>
      <ReactMarkdown source={state} />
    </div>
  )
}

Post.propTypes = {
  postId: PropTypes.string
}

export default Post
