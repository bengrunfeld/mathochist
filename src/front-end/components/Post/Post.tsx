import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import ReactMarkdown from 'react-markdown'
import getPostContent from '../../services/getPostContent'

import './styles.css'


const Post = ({postId}) => {
  const [state, setState] = useState('')

  // The function in useState only fires on the initial render.
  useEffect(() => {

    // Anon func that runs immediately
    (async() => 
      setState(await getPostContent(postId, setState)))()

  }, [postId])

  return (
    <div className='post col-lg-10'>
      <ReactMarkdown source={state} />
    </div>
  )
}

Post.propTypes = {
  postId: PropTypes.string
}

export default Post
