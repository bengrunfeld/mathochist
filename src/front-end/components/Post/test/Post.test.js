import React from 'react'
import Enzyme, { shallow, mount, render } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'

import Post from '../Post'


describe('<Post />', () => {
  test('matches the snapshot', () => {
    const tree = mount(<Post />)
    expect(tree).toMatchSnapshot()
  })
})
