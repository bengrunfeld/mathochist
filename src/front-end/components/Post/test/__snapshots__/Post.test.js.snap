// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> matches the snapshot 1`] = `
<Post>
  <div
    className="post col-lg-10"
  >
    <ReactMarkdown
      astPlugins={Array []}
      escapeHtml={true}
      plugins={Array []}
      rawSourcePos={false}
      renderers={Object {}}
      skipHtml={false}
      source=""
      sourcePos={false}
      transformLinkUri={[Function]}
    >
      <Root
        key="root-1-1"
      />
    </ReactMarkdown>
  </div>
</Post>
`;
